### PR TITLE
graph/{iterator,multi}: use map iterators for lines and weighted lines

### PR DESCRIPTION
**Pull request by kortschak**
Please take a look.

### CODE REVIEW

File: graph/iterator/lines_map.go [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+	pos   int
+	curr  graph.Line
+}
+
+// NewLines returns a Lines initialized with the provided lines, a
+// map of line IDs to graph.Lines. No check is made that the keys
+// match the graph.Line IDs, and the map keys are not used.
+//
+// Behavior of the Lines is unspecified if lines is mutated after
+// the call the NewLines.
```
Comment (vladimir-ch) [inline graph/iterator/lines_map.go:26]: s/the/to throughout

Comment (kortschak) [reply graph/iterator/lines_map.go]: Thanks. Fixed in other files not touched here too.

File: graph/iterator/map.go [hunk 2]
```diff
@@ -28,31 +28,44 @@ type emptyInterface struct {
...
 
 // newMapIterNodes returns a range iterator for a map of nodes.
+// The returned mapIter must not have its line or weightedLine methods called.
 func newMapIterNodes(m map[int64]graph.Node) *mapIter {
 	return &mapIter{m: eface(m)}
 }
 
 // newMapIterEdges returns a range iterator for a map of edges.
-// The returned mapIter must not have its node method called.
+// The returned mapIter must not have its node, line or weightedLine methodss called.
```
Comment (vladimir-ch) [inline graph/iterator/map.go:37]: ss

File: graph/iterator/map.go [hunk 3]
```diff
@@ -82,6 +95,28 @@ func (it *mapIter) node() graph.Node {
 	return *(*graph.Node)(mapiterelem(it.it))
 }
 
+// line returns the value of the iterator's current map entry.
+func (it *mapIter) line() graph.Line {
+	if it.it == nil {
+		panic("mapIter.node called before next")
```
Comment (vladimir-ch) [inline graph/iterator/map.go:101]: `mapIter.line` and similarly in `weightedLine`.

**Review by vladimir-ch — COMMENTED** [review]
Just nitpicks.

File: graph/iterator/map.go (hunk 2 above)
Comment (kortschak) [reply graph/iterator/map.go:37]: Done

File: graph/iterator/map.go (hunk 3 above)
Comment (kortschak) [reply graph/iterator/map.go:101]: Done. I'm looking forward to this being a single pair of generic functions.

**Review by vladimir-ch — APPROVED** [review]

**Pull request merged**